### PR TITLE
IAP doc: tutorial

### PR DESCRIPTION
**Pull request by xmo-odoo**
* fixed IO issue in iap/ (qweb returns bytes, but JSON needs text)
* added aliases in odoo.addons.iap for helpers (less annoying)
* added docstrings to charge's description & credit_template parameters
* added a basic technical tutorial
  - also an explanation as to how to use `quilt` to update patch queues, but maybe it should go in the doc readme since there are multiple such queues?

## ?
* needs way to easily test locally without access to test.odoo.com (e.g. third-party developers & al)
* unclear whether we're saying this should request apps or should work on any module?

## todo:
* Update technical docs (some stuff has changed since previous writeup)
* Better document exceptions serialisation stuff?

### CODE REVIEW

File: doc/webservices/iap.rst [hunk 1]
```diff
@@ -1,10 +1,41 @@
 :types: api
 
-
-:code-column:
-
 .. _webservices/iap:
 
+.. using sphinx-patchqueue:
```
Comment (mart-e) [inline doc/webservices/iap.rst]: is it a doc for those who wrote doc? 😉

Comment (xmo-odoo) [reply doc/webservices/iap.rst]: Yep, though maybe it should be in a doc/README file instead?

**Comment by xmo-odoo** [comment]
@rrahir 

* for the actual IAP, for service providers will the "portal" page (where they can create their services and manage their packs) be on odoo.com or iap.odoo.com?
* "My Account" > "Your InApp Services"? Are the sections normally prefixed with "My" or "Your" /cc @tde-banana-odoo

**Comment by rrahir** [comment]
@xmo-odoo 

-  The idea is to have a link 'in app sevices' in odoo.com that will redirect users on iap.odoo.com portal. where they will land on the 'my services' page (list of existing services). So technically, both ^^ But they will access this page via their odoo.com portal.

- Not sure for the section naming, the 'My'/ 'your' dilemma was present in saas-16 (I think), Think they removed he prefix now. I'm changing it in my internal branch. (should be on sandbox or test soon)

**Pull request merged**